kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -234,35 +237,34 @@ applyFirstSimplifierThatWorks
                         ++ show applicationResult
                         )
                     )
-                Monad.when
-                    (not (OrPattern.isFalse orRemainders)
+                if  not (OrPattern.isFalse orRemainders)
                     && not (acceptsMultipleResults multipleResults)
-                    )
-                    -- It's not obvious that we should accept simplifications
-                    -- that change only a part of the configuration, since
-                    -- that will probably make things more complicated.
-                    --
-                    -- Until we have a clear example that this can actually
-                    -- happen, we throw an error.
-                    ((error . show . Pretty.vsep)
-                        [ "Unexpected simplification result with remainder:"
-                        , Pretty.indent 2 "input:"
-                        , Pretty.indent 4 (unparse patt)
-                        , Pretty.indent 2 "results:"
-                        , (Pretty.indent 4 . Pretty.vsep)
-                            (unparse <$> Foldable.toList orResults)
-                        , Pretty.indent 2 "remainders:"
-                        , (Pretty.indent 4 . Pretty.vsep)
-                            (unparse <$> Foldable.toList orRemainders)
-                        ]
-                    )
-                return applicationResult
-        AttemptedAxiom.NotApplicable ->
-            applyFirstSimplifierThatWorks
-                evaluators
-                multipleResults
-                substitutionSimplifier
-                simplifier
-                axiomIdToSimplifier
-                patt
-                predicate
+                then
```

Comment:
I'd put the "do" after "then", in which case the "do" contents will be properly indented.

Also, I suggest that this log statement be triggered at most once per axiom id or something like that, otherwise it may be too noisy.

Reply:
That would be interesting feature request for the logger, something like `logOnce` which would take a `Hashable` as an identifier. I'll add a TODO here